### PR TITLE
feat:costcenter

### PR DESCRIPTION
**Pull request by xudaotutou**
### Summary
🚀🌐🛡️

This pull request adds a new frontend provider for the costcenter feature to the sealos application. It creates a new directory under `frontend/providers` with the necessary files and configurations for building, deploying, and running the costcenter frontend with next.js, chakra-ui, echarts, i18next, and sealos-desktop-sdk. It also extends the GitHub workflow for the frontend to include the costcenter provider and use the GitHub container registry as the image source.

> _We're building the `costcenter` frontend, me hearties, yo ho ho_
> _With next.js and chakra-ui, we'll make it nice and slow_
> _We'll dockerize and kubernetize, and push it to the cloud_
> _And sing this shanty all the way, we'll make our captain proud_

### Walkthrough
*  Add the costcenter frontend provider to the GitHub workflow for building, pushing, and deploying the frontend images ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-ffe3a297ed15e9a79846161a3731a580511bf13c6b5cc5a9e03ddcbd8515fdeeR53), [link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-ffe3a297ed15e9a79846161a3731a580511bf13c6b5cc5a9e03ddcbd8515fdeeR134))
*  Update the `deploy/cloud/init.sh` script to pull and save the costcenter frontend image from the GitHub container registry ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-7c20f8ecac334e472df2875627ce9de0e4297d7c839ef4ed0da3e80cb4e16595L10-R13), [link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-7c20f8ecac334e472df2875627ce9de0e4297d7c839ef4ed0da3e80cb4e16595R21))
*  Create a `.dockerignore` file for the costcenter frontend provider to exclude the `node_modules` directory from the docker build context ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-762ffc3e180d633a24a97aeb6f6f60faa8bd767b78739ec621746acbe27d178bL1-L-1))
*  Create a `.eslintrc.json` file for the costcenter frontend provider to extend the `next/core-web-vitals` eslint configuration for code quality and performance ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-964e70c937f21b5590f777bebd6a17b48ebe5ad635749620e770c68ccc442eeaR1-R3))
*  Create a `.gitignore` file for the costcenter frontend provider to ignore common files and directories that should not be tracked by git ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-cbae1f8f897edb736a5fe28859cec01eb1f48143f5ad770961de2983c2b61c1aR1-R36))
*  Create a `.prettierrc.js` file for the costcenter frontend provider to configure the prettier code formatter with some custom options ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-ee84aa9e374e2a4110c991cafc830bd422eb425bb755bb19c8664434c623792aL1-R19))
*  Create a `Dockerfile` for the costcenter frontend provider to build a production-ready docker image with the next.js application using a multi-stage build process ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-8d70eee56b73fdfdf440cadd2bb26d4f73cdb2b0fff200cb8437e4792a0a56b9R1-R75))
*  Create a `Makefile` for the costcenter frontend provider to define some common commands for building, running, and pushing the docker image ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-1be8c9091cbb09bd2a026e16395bad27b92bc469eee52b75e183cdc787759403L1-R45))
*  Create a `README.md` file for the costcenter frontend provider to document the directory structure, the build and push process, and some other relevant information ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-b43795e7799202cbb3ab8cc8642315d13918dbf95f79713dec613afc0a3a160aR1-R68))
*  Create a `Kubefile` for the costcenter frontend provider to define how to deploy the application to the sealos cloud cluster using the sealos registry and manifests ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-972fa290b24ab7df51ec4baf1aca3dbd614d867fa7140e1ce394dd174463b25aR1-R12))
*  Create a `appcr.yaml.tmpl` file for the costcenter frontend provider to define the app custom resource that registers the application to the sealos cloud applaunchpad ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-e4f0dd85bc6f34921be4de70c904e70b5c4ae4bafc918e56492b0c8e5e4a9dc1L1-R16))
*  Create a `deploy.yaml` file for the costcenter frontend provider to define the deployment, service, configmap, and namespace resources that create and expose the costcenter frontend pod ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-08d442378d1d5b9102a2d0eedfa18f9b02082912207c092c37baada608850aeaL1-R74))
*  Create a `ingress.yaml.tmpl` file for the costcenter frontend provider to define the ingress resource that routes the traffic to the costcenter frontend service ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-db5ed87d672b404e906bcad432ed6677fcdea619c3431803f9e9a1a2af822125R1-R53))
*  Create a `next-i18next.config.js` file for the costcenter frontend provider to configure the next-i18next library for internationalization ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-ef42e06d82344c4f1617b4f61a987244dc9bacb03ae8299ea8968da406416a45R1-R12))
*  Create a `next.config.js` file for the costcenter frontend provider to configure the next.js framework ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-18de783b6334529bd30e132a53a10248fa89ad606f806098d2b96c0be55969d5R1-R11))
*  Create a `package.json` file for the costcenter frontend provider to define the project metadata, dependencies, and scripts ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-34d38e3d749383a68496dd6da64c7aa812c2bb675609691dbc1b9943d4d23da6L1-R55))
*  Create a `common.json` file for the costcenter frontend provider to store the English translation strings for the common UI elements and texts ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-22a0af9ffd00ea2fcfdc0db3fb38d437eb615c51f91a38ba0c3c100b20a09ddbL1-R49))
*  Create a `common.json` file for the costcenter frontend provider to store the Chinese translation strings for the common UI elements and texts ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-54169e3c66c9921e2d46d2574fd77362a7f5e405e054acb1938dcf5c275e6c83L1-R49))
*  Create a `index.tsx` file for the costcenter frontend provider to implement a language selector component that allows the user to switch between English and Chinese languages ([link](https://github.com/labring/sealos/pull/3326/files?diff=unified&w=0#diff-65b1ea1bb8d5c044153421c10315ced9bad5bb8f546d985a4b07f278ee0334e5R1-R60))

### CODE REVIEW

File: frontend/providers/costcenter/deploy/manifests/appcr.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: app.sealos.io/v1
+kind: App
+metadata:
+  name: costcenter
+  namespace: app-system
+spec:
+  data:
+    desc: sealos CLoud costcenter
+    url: "https://costcenter.{{ .cloudDomain }}"
+  icon: "https://costcenter.{{ .cloudDomain }}/logo.svg"
```
Comment (lingdie) [inline frontend/providers/costcenter/deploy/manifests/appcr.yaml.tmpl:10]: icon这个地址对吗？ @zjy365

File: deploy/cloud/init.sh [hunk 2]
```diff
@@ -7,13 +7,16 @@ sealos pull ghcr.io/labring/sealos-cloud-desktop-frontend:dev
 sealos pull ghcr.io/labring/sealos-cloud-terminal-frontend:dev
 sealos pull ghcr.io/labring/sealos-cloud-applaunchpad-frontend:dev
 sealos pull ghcr.io/labring/sealos-cloud-dbprovider-frontend:dev
+sealos pull ghcr.io/labring/sealos-cloud-costcenter-frontend:dev
```
Comment (lingdie) [inline deploy/cloud/init.sh:10]: 这里可以先不加上

File: frontend/providers/costcenter/deploy/manifests/appcr.yaml.tmpl (hunk 1 above)
Comment (zjy365) [reply frontend/providers/costcenter/deploy/manifests/appcr.yaml.tmpl:10]: > 
这个要看 @xudaotutou  传logo没有，没有logo的话问ui要一下

**Review by zzjin — APPROVED** [review]

File: frontend/providers/costcenter/src/pages/api/account/getAmount.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const account_meta: CRDMeta = {
+      group: 'account.sealos.io',
+      version: 'v1',
+      namespace: 'sealos-system',
+      plural: 'accounts'
+    };
+
+    const accountDesc = await GetCRD(kc, account_meta, user.name);
+    if (accountDesc !== null && accountDesc.body !== null && accountDesc.body.status !== null) {
+      const accountStatus = accountDesc.body.status as accountStatus;
```
Comment (c121914yu) [inline frontend/providers/costcenter/src/pages/api/account/getAmount.ts:30]: At least one response must be guaranteed。

if(){....}
else {throw new Error()}

**Review by c121914yu — APPROVED** [review]

**Pull request merged**